modules/tinymce/src/core/test/ts/browser/content/EditorContentTest.ts

```diff
@@ -92,17 +89,19 @@ describe('browser.tinymce.core.content.EditorContentTest', () => {
   // TODO: TINY-8367 The table plugin code has been moved to core so there is now always an extra DIV in the dom for the table resize bars
   // Safari differs in behaviour compared to the other browsers when getting text content for inline mode.
   // When the resize bar div is included in the DOM, editor.getBody().innerText includes two extra \n at the end
-  it('TINY-6281: getContent text', () => testGetTextContent('<p>Text to be retrieved</p>', 'Text to be retrieved' + (isSafari() ? '\n\n' : '')));
+  it('TINY-6281: getContent text', () => testGetTextContent('<p>Text to be retrieved</p>', '\n\nText to be retrieved\n\n'));
+
+  it('TINY-8578: getContent text, empty line in div', () => testGetTextContent('<div><p></p></div>', '\n\n\n\n\n\n\n'));
 
-  it('TINY-8578: getContent text, empty line in div', () => testGetTextContent('<div><p></p></div>', ''));
+  it('TINY-8578: getContent text, empty line', () => testGetTextContent('<p></p>', '\n\n\n\n\n'));
 
-  it('TINY-8578: getContent text, empty line', () => testGetTextContent('<p></p>', ''));
+  it('TINY-8578: getContent text, two empty lines in div', () => testGetTextContent('<div><p></p><p></p></div>', '\n\n\n\n\n\n\n\n\n\n'));
 
-  it('TINY-8578: getContent text, two empty lines in div', () => testGetTextContent('<div><p></p><p></p></div>', isSafari() ? '\n\n' : '\n\n\n\n'));
+  it('TINY-8578: getContent text, repeating two empty lines in divs', () => testGetTextContent('<div><p></p><p></p></div><div><p></p><p></p></div>', '\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n'));
 
-  it('TINY-8578: getContent text, two empty lines', () => testGetTextContent('<p></p><p></p>', isSafari() ? '\n\n' : '\n\n\n\n'));
+  it('TINY-8578: getContent text, two empty lines', () => testGetTextContent('<p></p><p></p>', '\n\n\n\n\n\n\n\n'));
 
-  it('TINY-6281: getContent text with empty editor', () => testGetTextContent('', ''));
+  it('TINY-6281: getContent text with empty editor', () => testGetTextContent('', '\n\n\n\n\n'));
```

Comment:
The new method have added a lot of newlines everywhere. While this is now the same between the browsers, do we want to continue in this direction or try a different one?

Reply:
Yeah I agree with Spocke here... the TBIO solution is not going to work for TinyMCE as it doesn't cover a lot of cases so it'll be a regression in functionality. It also doesn't have context of the schema nor things we generally need to ignore like bogus elements. If we need a custom walker solution then we are going to have to expand on that quite a bit, implement our own (as we maybe can't change the TBIO one depending on where and how it's being used) or look at a third-party library for this. Both are going to be a very large task.

Reply:
It was never going to be a silver bullet. I'd be surprised if the TBIO logic is used anywhere that would have a problem with us adjusting the newline behaviour, if it's even used at all.

I do think a custom serializer is the correct solution because at least then it's the same for all browsers. But I guess if we can find a workaround that will solve the customer issue we can schedule something larger down the track.

Reply:
I wonder if we could do something like this at least until we can make a proper get text mode:
```
body = editor.getBody();
clonedBody = body.cloneNode(false);
clonedBody.innerHTML = editor.getContent();
body.parentNode.insertBefore(clonedBody, body);
text = clonedBody.innerText;
clonedBody.remove();
console.log(text);
```
Would not be efficient but it would remove any bogus nodes and also render that in context since I think a detached body might do very strange things when it doesn't know how to render things. Does it use the default user agent css. Seems safer to stick next to the body element of the editor.